### PR TITLE
generate mailer layout files if it does not already exist

### PR DESCRIPTION
**Pull request by y-yagi**
Currently, if `ApplicationMailer` does not exist, it is generated when
run the mailer generator, but layouts files does not generate.
However, because it uses the layouts in `ApplicationMailer`,
layouts are required.

Follow up to #24161

### CODE REVIEW

File: railties/lib/rails/generators/erb/mailer/templates/layout.html.erb.tt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <style>
+      /* Email styles need to be inline */
+    </style>
```
Comment (eileencodes) [inline railties/lib/rails/generators/erb/mailer/templates/layout.html.erb.tt]: I don't think adding empty style tags is necessary.

File: railties/test/generators/plugin_generator_test.rb [hunk 2]
```diff
@@ -669,6 +669,20 @@ def test_generate_application_mailer_when_does_not_exist_in_mountable_engine
...
 
+  def test_generate_mailer_layouts_when_does_not_exist_in_mountable_engine
+    run_generator [destination_root, '--mountable']
+    capture(:stdout) do
+      `#{destination_root}/bin/rails g mailer User`
+    end
+
+    assert_file "#{destination_root}/app/views/layouts/bukkits/mailer.text.erb" do |view|
+      assert_match(/<%= yield %>/, view)
+    end
```
Comment (eileencodes) [inline railties/test/generators/plugin_generator_test.rb]: Can you add a space between these 2 asserts? 

```ruby
assert_file "#{destination_root}/app/views/layouts/bukkits/mailer.text.erb" do |view|
  assert_match(/<%= yield %>/, view)
end

assert_file "#{destination_root}/app/views/layouts/bukkits/mailer.html.erb" do |view|
  assert_match(%r{<body>\n    <%= yield %>\n  </body>}, view)
end
```

File: railties/lib/rails/generators/erb/mailer/templates/layout.html.erb.tt (hunk 1 above)
Comment (y-yagi) [inline railties/lib/rails/generators/erb/mailer/templates/layout.html.erb.tt]: The layout file of new rails app contains empty style tags. 
https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/templates/app/views/layouts/mailer.html.erb.tt#L5..L7 

Should not make it the same?

File: railties/test/generators/plugin_generator_test.rb (hunk 2 above)
Comment (y-yagi) [inline railties/test/generators/plugin_generator_test.rb]: Sure!  I updated.

File: railties/lib/rails/generators/erb/mailer/templates/layout.html.erb.tt (hunk 1 above)
Comment (eileencodes) [inline railties/lib/rails/generators/erb/mailer/templates/layout.html.erb.tt]: That's fine then, I didn't see that one. 👍

**Pull request merged**